docs/source/developers/python.rst

```diff
@@ -636,6 +636,52 @@ Caveats
 
 The Plasma component is not supported on Windows.
 
+Deleting stale build artifacts
+==============================
+
+When there have been changes to the structure of the Arrow C++ library or Pyarrow,
+a thorough cleaning is recommended as a first attempt to fixing build errors.
+
+.. note::
+
+   It is not necessarily intuitive from the error itself that the problem is due to stale artifacts.
+   Example of a build error from stale artifacts is "Unknown CMake command "arrow_keep_backward_compatibility".
+
+To delete stale Arrow C++ build artifacts:
+
+.. code-block::
+
+  $ rm -rf arrow/cpp/build
+
+To delete stale PyArrow build artifacts:
+
+.. code-block::
+
+  $ pushd python
+  $ python3 setup.py clean
+  $ popd
+
+To delete misc build artifacts that are hidden from ``git``:
+
+.. code-block::
+
+  $ git clean -Xfd .
+
+If using a Conda environment, there are some build artifacts that get installed in
+``$ARROW_HOME`` (aka ``$CONDA_PREFIX``). For example, ``$ARROW_HOME/lib/cmake/Arrow*``.
```

Comment:
We may want to remove `$ARROW_HOME/include/{arrow,gandiva,plasma}`, `$ARROW_HOME/lib/lib{arrow,gandiva,plasma}*` and so on but we will not be able to maintain the file lists to be removed...

So recreating a new Conda environment will be better.

Reply:
That is why I provided that recommendation! It is a bigger thing to remove and reinstall, but it makes for cleaner, simpler advice.

Reply:
I think it is useful to mention this at least that you can clean manually, as is done right now. The last weeks I had to remove the cmake files several times (also from switching between (older) branches); if I would have created a new conda env each time, that would have become annoying.